docs-starlight/src/content/docs/04-reference/03-strict-controls.mdx

```diff
@@ -141,6 +143,10 @@ Throw an error when using the `validate-all` command.
 
 Disable reading of dependency inputs to enhance dependency resolution performance by preventing recursively parsing Terragrunt inputs from dependencies.
 
+<Aside type="tip">
+Skipping dependency inputs is a performance optimization. For more details on performance optimizations, their tradeoffs, and other performance tips, read the dedicated [Performance documentation](/docs/troubleshooting/performance).
```

Comment:
as a user reading this, the question that comes to mind is "So why not have it always disabled?"

Reply:
In the performance documentation docs, I explicitly say basically "you should definitely always have this disabled"